src/transformStateWithClones.js

```diff
@@ -7,7 +7,32 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const result = [];
+  let currentState = { ...state };
+
+  for (const action of actions) {
+    if (action.type === 'addProperties') {
+      const nextState = {
+        ...currentState, ...action.extraData,
+      };
+
+      result.push(nextState);
+      currentState = nextState;
+    } else if (action.type === 'removeProperties') {
+      const nextState = { ...currentState };
+
+      for (const key of action.keysToRemove) {
+        delete nextState[key];
+      }
+      result.push(nextState);
+      currentState = nextState;
+    } else if (action.type === 'clear') {
+      result.push({});
+      currentState = {};
+    }
+  }
```

Comment:
Good work but let's rewrite it to switch/case statement since we rely on a single action.type value